libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp

```diff
@@ -369,12 +369,14 @@ bool ESP8266WiFiSTAClass::getAutoReconnect() {
  * returns the status reached or disconnect if STA is off
  * @return wl_status_t
  */
-uint8_t ESP8266WiFiSTAClass::waitForConnectResult() {
+uint8_t ESP8266WiFiSTAClass::waitForConnectResult(uint32_t wait_secs) {
     //1 and 3 have STA enabled
     if((wifi_get_opmode() & 1) == 0) {
         return WL_DISCONNECTED;
     }
-    while(status() == WL_DISCONNECTED) {
+    int i = 0; 
+    int try_times = wait_secs*10;
+    while((!status() || status() >= WL_DISCONNECTED) && ((wait_secs==0)? true : (i++ < try_times))) {

```

Comment:
This line is difficult to read.
* `!status()` tests whether the status is non-zero, but status is an enum and it’s unobvious which value is zero.
* The part after && is really complicated.

How about:
```
for (int timeout = wait_secs * 10; wait_secs == 0 || timeout > 0; --timeout) {
  auto st = status();
  switch (st) {
    case WL_CONNECTED:
    case WL_NO_SSID_AVAIL:
    case WL_CONNECT_FAILED:
      return st;
    default:
      delay(100);
      break;
  }
}

Reply:
Agree with @yoursunny . However:

I have a PR with a class that encapsulates the polled timeout. I suggest waiting a bit until it's merged, then using it, instead of reinventing the wheel for the nth time.
the switch should define cases for all enum values, and leave the default case for values not in the enum (error).